### PR TITLE
Fixes TOC in dockerfile_best_practices.md

### LINKED ISSUE #14573: Documentation: TOC links in Best Practices page links to reference instead of the heading
https://docs.docker.com/articles/dockerfile_best-practices

To the right of the page is a TOC listing the major headings "On this page:". The first half of the TOC is ok. The bottom half includes the best practices for commands (FROM, RUN, CMD, ... etc) and these links to the references. eg. https://docs.docker.com/reference/builder/#from instead of the actual heading/section on the page.

This behaviour makes it difficult to navigate and jump to the specific heading/section by using the TOC.

### PR DESCRIPTION
**Pull request by vdemeester**
Decided to put the link in a reference word like : `RUN` ([reference](...))

Closes #14573

Signed-off-by: Vincent Demeester vincent@sbr.pm

### CODE REVIEW

File: docs/articles/dockerfile_best-practices.md [hunk 1]
```diff
@@ -254,7 +254,7 @@ Similar to having constant variables in a program (as opposed to hard-coding
 values), this approach lets you change a single `ENV` instruction to
 auto-magically bump the version of the software in your container.
 
-### [`ADD`](https://docs.docker.com/reference/builder/#add) or [`COPY`](https://docs.docker.com/reference/builder/#copy)
+### `ADD` ([reference](https://docs.docker.com/reference/builder/#add) or [`COPY`](https://docs.docker.com/reference/builder/#copy))
```
Comment (thaJeztah) [inline docs/articles/dockerfile_best-practices.md]: Add a "reference" link after `COPY` as well, to be consistent

Comment (vdemeester) [inline docs/articles/dockerfile_best-practices.md]: @thaJeztah oh right, missed it :blush:

**Comment by thaJeztah** [comment]
Looks like you missed `FROM`; https://github.com/docker/docker/pull/14581/files#diff-938bac722370ef89c88729e30566b506R129

**Comment by vdemeester** [comment]
@thaJeztah yeah I did :sweat:.

**Comment by thaJeztah** [comment]
@vdemeester LOL, no problem, was still looking a bit. Looking at this page, I don't really like the use of back-ticks (code-blocks) in headings, perhaps we should just change to regular headings.

I think adding a separate "reference" link makes sense (not really a fan of using the heading itself as a link). Wondering if we should put that link *below* the headings (but this is just thinking out loud here), for example:


### EXPOSE

*[Dockerfile reference for the EXPOSE instruction](https://docs.docker.com/reference/builder/#expose)*

The `EXPOSE` instruction indicates the ports on which a container will listen for ....

**Comment by vdemeester** [comment]
@thaJeztah Sounds reasonable to me, and it will look better. I'll give it a try :wink:.

**Comment by thaJeztah** [comment]
Thanks, @vdemeester, I like that change! Here's a screenshot how it looks with this PR;

screenshot



While looking at this PR, I noticed there's a stray `#onbuild` anchor in the link at the bottom; https://github.com/docker/docker/pull/14581/files#diff-938bac722370ef89c88729e30566b506R464. Could you remove that as well? :heart:

LGTM

ping @moxiegirl wdyt?

**Comment by vdemeester** [comment]
@thaJeztah will do :-)

**Comment by moxiegirl** [comment]
@vdemeester Thanks for the catches and the fixes.  I noticed the entry point link is not working. Do you want to get that one too?

https://docs.docker.com/reference/builder/#entrypoint

**Comment by vdemeester** [comment]
@moxiegirl sure thing :wink:.

**Comment by vdemeester** [comment]
@moxiegirl and done I guess :blush:.

**Comment by moxiegirl** [comment]
Saweet. Thank you @vdemeester LGTM

**Comment by thaJeztah** [comment]
Thanks Vincent!

**Pull request merged**